File: iOS-ObjC/CustomStyle.m
#import "CustomStyle.h"
@import Hex;
@import FormTextField;

@implementation CustomStyle

+ (void)apply {
    UIColor *enabledBackgroundColor = [UIColor whiteColor];
    UIColor *enabledBorderColor = [[UIColor alloc] initWithHex:@"DFDFDF"];
    UIColor *enabledTextColor = [[UIColor alloc] initWithHex:@"455C73"];
    UIColor *activeBorderColor = [[UIColor alloc] initWithHex:@"70D7FF"];

    [[FormTextField appearance] setBorderWidth:2];
    [[FormTextField appearance] setCornerRadius:10];
    [[FormTextField appearance] setAccessoryButtonColor:activeBorderColor];
    [[FormTextField appearance] setFont:[UIFont fontWithName:@"AvenitNext-Regular" size:15]];

    [[FormTextField appearance] setEnabledBackgroundColor:enabledBackgroundColor];
    [[FormTextField appearance] setEnabledBorderColor:enabledBorderColor];
    [[FormTextField appearance] setEnabledTextColor:enabledTextColor];

    [[FormTextField appearance] setValidBackgroundColor:enabledBackgroundColor];
    [[FormTextField appearance] setValidBorderColor:enabledBorderColor];
    [[FormTextField appearance] setValidTextColor:enabledTextColor];

    [[FormTextField appearance] setActiveBackgroundColor:enabledBackgroundColor];
    [[FormTextField appearance] setActiveBorderColor:activeBorderColor];
    [[FormTextField appearance] setActiveTextColor:enabledTextColor];

    [[FormTextField appearance] setInactiveBackgroundColor:enabledBackgroundColor];
    [[FormTextField appearance] setInactiveBorderColor:enabledBorderColor];
    [[FormTextField appearance] setInactiveTextColor:enabledTextColor];

    [[FormTextField appearance] setDisabledBackgroundColor:[[UIColor alloc] initWithHex:@"DFDFDF"]];
    [[FormTextField appearance] setDisabledBorderColor:[[UIColor alloc] initWithHex:@"DFDFDF"]];
    [[FormTextField appearance] setDisabledTextColor:[UIColor whiteColor]];

    [[FormTextField appearance] setInvalidBackgroundColor:[[UIColor alloc] initWithHex:@"FFC9C8"]];
    [[FormTextField appearance] setInvalidBorderColor:[[UIColor alloc] initWithHex:@"FF4B47"]];
    [[FormTextField appearance] setInvalidTextColor:[[UIColor alloc] initWithHex:@"FF4B47"]];
}

@end

File: Pod/App/CustomStyle.swift
import Foundation
import Hex

struct CustomStyle {
    static func apply() {
        TextField.appearance().borderWidth = 2
        TextField.appearance().cornerRadius = 10
        TextField.appearance().customFont = UIFont(name: "AvenirNext-Regular", size: 15)!

        TextField.appearance().activeBackgroundColor = UIColor.whiteColor()
        TextField.appearance().activeBorderColor = UIColor(hex: "70D7FF")

        TextField.appearance().validBackgroundColor = UIColor.whiteColor()
        TextField.appearance().validBorderColor = UIColor(hex: "70D7FF")

        TextField.appearance().enabledBackgroundColor = UIColor.whiteColor()
        TextField.appearance().enabledBorderColor = UIColor(hex: "DFDFDF")
        TextField.appearance().enabledTextColor = UIColor(hex: "455C73")

        TextField.appearance().disabledBackgroundColor = UIColor(hex: "DFDFDF")
        TextField.appearance().disabledBorderColor = UIColor(hex: "DFDFDF")
        TextField.appearance().disabledTextColor = UIColor.grayColor()

        TextField.appearance().inactiveBackgroundColor = UIColor.whiteColor()
        TextField.appearance().inactiveBorderColor = UIColor(hex: "DFDFDF")

        TextField.appearance().invalidBackgroundColor = UIColor(hex: "FFC9C8")
        TextField.appearance().invalidBorderColor = UIColor(hex: "FF4B47")

        TextField.appearance().accessoryButtonColor = UIColor(hex: "70D7FF")
    }
}




